tests/integration-tests/tests/monitoring/test_monitoring.py

```diff
@@ -190,3 +195,15 @@ def _verify_alarms(alarms, metric_name, instance_id):
         assert_that(alarms[0]["Dimensions"]).contains({"Name": "path", "Value": "/"}).contains(
             {"Name": "InstanceId", "Value": instance_id}
         )
+
+
+def _test_detailed_monitoring(ec2_client, compute_instance_ids):
+    ec2_response = ec2_client.describe_instances(InstanceIds=compute_instance_ids)
+    monitoring_states = [
+        instance.get("Monitoring").get("State")
+        for reservation in ec2_response.get("Reservations")
+        for instance in reservation.get("Instances")
+    ]
+    assert_that(monitoring_states).is_not_empty()
+    assert_that(set(monitoring_states)).is_length(1)
+    assert_that(monitoring_states[0]).is_equal_to("enabled")
```

Comment:
This can be simplified to:
```
assert_that(ec2_response["Reservations"][0]["Instances"][0]["Monitoring"]["State"]).is_equal_to(""enabled"")
```

Reply:
Can we guarantee there is only 1 compute node? I wrote the function to work with any number of compute nodes and check they are all in the "enabled" state.

Reply:
In your code, you are asserting `assert_that(set(monitoring_states)).is_length(1)`, doesn't that mean the function only works for one compute node?

Reply:
Not necessarily. There can be any non-zero number of compute nodes and here I'm checking that (1) they all have the same monitoring state; (2) that state is the expected value (here it's "enabled").